plumbing/format/packfile/parser.go

```diff
@@ -234,6 +234,15 @@ func (p *Parser) indexObjects() error {
 				return err
 			}
 
+			// move children of placeholder parent into actual parent in case this was
+			// a non-external delta reference
+			if placeholder, ok := p.oiByHash[sha1]; ok {
+				ota.Children = placeholder.Children
+				for _, c := range ota.Children {
```

Comment:
Will children have different parents?
if not, we can exist the loop on first element

Reply:
@sarabala1979 The goal here is to re-assign the parent of all the children, I think we cannot exit the for loop after first element.